doc/specs/#364 - Feature Toggle.md

```diff
@@ -0,0 +1,80 @@
+---
+author: John McPherson @JohnMcPMS
+created on: 2020-06-04
+last updated: 2020-06-04
+issue id: 364
+---
+
+# Feature Toggle
+
+For [#364](https://github.com/microsoft/winget-cli/issues/364)
+
+## Abstract
+
+As features are implemented within winget, they may cause disruption to users oblivious to the fact that they are in progress. In order to allow work to be done in master, and distributed to early adopters for their feedback, this spec suggests that settings should be used to control "experimental" features. We use this term [as others have](https://github.com/PowerShell/PowerShell-RFC/blob/master/5-Final/RFC0029-Support-Experimental-Features.md) to mean work in progress that is not yet ready for relase to the general audience.
+
+## Inspiration
+
+We realize that we need to be able to:
+1. deliver work in progress to users so that we can receive feedback
+2. work in master without creating multiple, divergent branches
+3. not disrupt users who do not wish to investigate unfinished features
+
+Thank you to @megamorf for the link to the PowerShell Experimental Features RFC.
+
+## Solution Design
+
+A new settings block will be added to the settings file for experimental features to be toggled. By default, all experimental features will be disabled.
+
+A hidden command will be added to the root:
+```
+>winget features
+The following experimental features are in progress.
+They can be configured through the settings file (winget settings).
+
+Feature  Status   Link
+--------------------------------------------------------------------
+Example1 Enabled  https://github.com/microsoft/winget-cli/issues/364
+Example2 Disabled https://github.com/microsoft/winget-cli/issues/396
+```
+When run, this command will give the status of each feature (enabled/disabled), as well as a link to the Issue or Spec.
+
+Internally, a single flag enum will be used to reference experimental features. All command line parsing objects will have a field added for the feature(s) to which they belong. Any internal behavioral changes will be made based on a check using this enum. In this way, transitioning a feature either to release or remove can be done by finding by a single identifier.
+
```

Comment:
We may want to consider a "global" revert back to stable like `winget features --stable` or `winget features --enable` and the complimentary `winget features --disable`. I could go either way with this.

Reply:
What does stable means? It seems to me that stable means no experimental features, which means disable. 

What does enable means? Is it turn all experimental features?

Reply:
There's also `stable` ambiguity when you start rolling out A/B testing.